front/src/components/CTF/CardList.vue

```diff
@@ -1,58 +1,208 @@
 <template>
-  <div class="row q-col-gutter-lg justify-evenly">
-    <div
-      v-for="(column, idx) in columns"
-      :key="idx"
-      class="column col q-gutter-lg justify-start align-center"
+  <div class="row justify-center scroll-anchor">
+    <q-btn color="primary" @click="fetchMore(shortDate.prev(ctfByMonth[0]))">
+      Load previous ctfs
+    </q-btn>
+  </div>
+  <div
+    v-for="group of ctfByMonth"
+    :key="shortDate.toId(group)"
+    class="full-size"
+    :data-date="shortDate.toId(group)"
+  >
+    <IntersectionObserver
+      :options="{ rootMargin: '200px' }"
+      class="column gap-lg"
+      @show.once="fetchMoreCtfs(group)"
     >
-      <q-intersection
-        v-for="ctf in column"
+      <div class="row items-center gap-md scroll-anchor">
+        <q-separator class="col" />
+        <div class="col-auto col text-h6 q-py-sm">
+          {{
+            date
+              .buildDate({ year: group.year, month: group.month })
+              .toLocaleDateString(undefined, { year: 'numeric', month: 'long' })
+          }}
+        </div>
+        <q-separator class="col" />
+      </div>
+      <div
+        v-for="ctf of group.ctfs"
         :key="ctf.nodeId"
-        once
-        style="min-height: 550px"
-        transition="fade"
+        class="ctf-card-container"
       >
-        <div>
-          <card :ctf="ctf" />
-        </div>
-      </q-intersection>
-    </div>
+        <ctf-card :ctf="ctf" :data-ctf-id="ctf.id" class="ctf-card" />
+      </div>
+      <q-card
+        v-if="group.ctfs.length === 0"
+        :key="shortDate.toId(group)"
+        class="full-size column items-center justify-center no-ctf"
+      >
+        <q-card-section>
+          <div class="text-center text-h5 q-pt-md">No CTFs this month</div>
+        </q-card-section>
+        <q-card-section class="text-center">
+          <q-icon name="hotel" size="128px" />
+        </q-card-section>
+      </q-card>
```

Comment:
I think this `q-card-section` is a bit too big. On mobile it will occupy a full screen to display that there is no information available. On desktop it is also quite large. I would suggest to lower the size and padding so more information would fit on one screen.